lib/rubocop/cop/lint/duplicate_regexp_character_class_element.rb

```diff
@@ -32,35 +32,53 @@ def on_regexp(node)
           end
         end
 
-        # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
         def each_repeated_character_class_element_loc(node)
           node.parsed_tree&.each_expression do |expr|
             next if skip_expression?(expr)
 
             seen = Set.new
-            enum = expr.expressions.to_enum
-            expression_count = expr.expressions.count
+            group_expressions(node, expr.expressions) do |group|
+              group_source = group.map(&:to_s).join
 
-            expression_count.times do |current_number|
-              current_child = enum.next
-              next if within_interpolation?(node, current_child)
+              yield source_range(group) if seen.include?(group_source)
 
-              current_child_source = current_child.to_s
-              next_child = enum.peek if current_number + 1 < expression_count
+              seen << group_source
+            end
+          end
+        end
+
+        private
 
-              if seen.include?(current_child_source)
-                next if start_with_escaped_zero_number?(current_child_source, next_child.to_s)
+        def group_expressions(node, expressions)
+          expressions = expressions.to_a
 
-                yield current_child.expression
-              end
+          until expressions.empty?
+            group = [expressions.shift]
+            next if within_interpolation?(node, group.first)
 
-              seen << current_child_source
-            end
+            pop_octal_digits(group, expressions) if escaped_octal?(group.first.to_s)
```

Comment:
How about leaving a comment like this?

```suggestion
            # Compatible logic for regexp_parser < 2.7.
            pop_octal_digits(group, expressions) if escaped_octal?(group.first.to_s)
```